rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/model/ParametricFileFacts.java

```diff
@@ -130,11 +138,21 @@ private void reportTypeCheckerMessages(List<Diagnostic> messages) {
 
         public void invalidate(boolean isClosing) {
             summary.invalidate(isClosing);
+            lastCalculateId.incrementAndGet(); // All scheduled calls to calculate are now stale
```

Comment:
Check  #355 every change has a version, can we not use that for this feature?

Reply:
Temporarily added a comment why using version numbers is complicated in the current one-summary approach. After implementing the two-summaries approach, this complication should be gone, so using version numbers should be ok.